### PR TITLE
Start using concurrent-skiplist

### PR DESCRIPTION
**Pull request by osa1**
Implements #103. DON'T MERGE YET, `concurrent-skiplist` also needs to be updated, some internal modules are now used by `lvish`, and some dependencies need to be relaxed.

### CODE REVIEW

**Comment by osa1** [comment]
OK, benchmark program is now fixed on all platforms.

Question: Is there a point in running benchmarks on travis? I think we should keep building benchmark program but there's no point in running it. If you're OK with that I'll remove relevant line from travis file.

**Comment by rrnewton** [comment]
Yeah, I don't think benchmarking on Travis is needed, although it may be a sort of statistical curiosity as discussed in [this issue](https://github.com/travis-ci/travis-ci/issues/352).

(Incidentally, that thread also mentions http://openbenchmarking.org/, which looks pretty interesting.)

It says "Don't merge yet" at the top, so even though this passed on Travis I'll hold off and let you merge it when ready.

File: .travis.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ before_install:
  # Use sed to replace the SSH URL with the public URL, then initialize submodules
  - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules
  - git submodule update --init
+ - git clone https://github.com/osa1/concurrent-skiplist -b refactoring ./haskell/concurrent-skiplist
```
Comment (osa1) [inline .travis.yml]: The reason I'm blocking this pull request is this line: I'm currently using my fork of `concurrent-skiplist`. I added this line to test travis builds. @rrnewton I have a pull request in `concurrent-skiplist`, if you merge that I can change this line and I think this is ready to merge.

Comment (rrnewton) [inline .travis.yml]: Ok, I added you as a contributor to my copy of the repo -- so merge PRs or push at will.

Also, let's switch that to a proper submodule.  Or, if we ever put a direct `git clone` hack in a script, it should be fixed to a specific SHA.  (Which effectively simulates a submodule.)  Grabbing the tip of a branch is just.... icky.

**Comment by lkuper** [comment]
I agree with Ryan that benchmarking on Travis doesn't really make sense, although I like the idea of continuing to *build* the benchmark executable on Travis.

**Pull request merged**